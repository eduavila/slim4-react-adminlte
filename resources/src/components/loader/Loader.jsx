import React from 'react';

import './Loader.scss';

export default props => {
    return (
        <div className={`ui small segment ${ props.content ? ` content-preloader`:``}` }>
            <div className="ui active inverted dimmer">
                <div className={`ui loader ${ !props.notViewDescription ? 'text':'' } ${ props.size }`}>
                    { !props.notViewDescription ? ( props.description || 'Carregando..') : null }
                </div>
            </div>
        </div>
    )
}
