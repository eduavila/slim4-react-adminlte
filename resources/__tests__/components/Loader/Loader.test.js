import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

import Loader from '../../../src/components/loader/Loader';

describe('<Loader />', () => {
    describe('render()', () => {
        test('renders the component', () => {
            const wrapper = shallow(<Loader />);
        
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        test('renders com descrição', () => {
            const wrapper = shallow(<Loader description="Carregando novo registro"/>);
        
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        test('renders sem texto', () => {
            const wrapper = shallow(<Loader notViewDescription/>);
           
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});