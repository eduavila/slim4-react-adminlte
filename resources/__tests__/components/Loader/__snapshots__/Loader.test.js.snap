// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> render() renders com descrição 1`] = `
<div
  className="ui small segment "
>
  <div
    className="ui active inverted dimmer"
  >
    <div
      className="ui loader text undefined"
    >
      Carregando novo registro
    </div>
  </div>
</div>
`;

exports[`<Loader /> render() renders sem texto 1`] = `
<div
  className="ui small segment "
>
  <div
    className="ui active inverted dimmer"
  >
    <div
      className="ui loader  undefined"
    />
  </div>
</div>
`;

exports[`<Loader /> render() renders the component 1`] = `
<div
  className="ui small segment "
>
  <div
    className="ui active inverted dimmer"
  >
    <div
      className="ui loader text undefined"
    >
      Carregando..
    </div>
  </div>
</div>
`;
